### PR TITLE
Add README, Add Ray cluster code

### PR DESCRIPTION
**Pull request by Joeavaikath**
Added README file.
Added Ray cluster connection code.
Added daily compaction code using dateutil package.

### CODE REVIEW

File: Ray-Only.ipynb [hunk 1]
```diff
@@ -1,5 +1,21 @@
 {
```
Comment (rimolive) [inline Ray-Only.ipynb:1]: <p>What if instead to <code>avoid problems</code> just <code>to load the new libraries</code>?  <span class="ql-cursor">﻿</span></p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/AICoE/ray-telemetry-compaction/pull/4/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='5'/>

File: Ray-Only.ipynb [hunk 2]
```diff
@@ -46,70 +86,459 @@
...
+   "source": [
+    "#%%capture output \n",
+    "@ray.remote\n",
+    "def compaction():\n",
+    "    firstDay = date(2021,1,1)\n",
+    "    lastDay = firstDay + relativedelta(months = 1)\n",
+    "    #print(lastDay, firstDay)\n",
+    "    duration = lastDay - firstDay\n",
+    "    #print(duration)\n",
+    "    InitializeS3()\n",
```
Comment (gmfrasca) [inline Ray-Only.ipynb:423]: where is this coming from?

File: Ray-Only.ipynb [hunk 3]
```diff
@@ -46,70 +86,459 @@
...
+    "jupyter": {
+     "source_hidden": true
+    },
+    "tags": []
+   },
+   "outputs": [],
+   "source": [
+    "firstDay = date(2021, 1, 1)\n",
+    "lastDay = firstDay + relativedelta(months = 1)\n",
+    "print(lastDay, firstDay)\n",
```
Comment (gmfrasca) [inline Ray-Only.ipynb]: there are a lot of these print statements with no context, which may be useful for debugging but is hard to decipher for other users who are just reading result cells' messages.  Could you either add some accompanying text to define what these messages are for, or remove them if they were only used for development/debugging?

File: Ray-Only.ipynb [hunk 4]
```diff
@@ -46,70 +86,459 @@
...
+    "duration = lastDay - firstDay\n",
+    "print(duration)\n",
+    "for _ in range(12):\n",
+    "    for i in range(duration.days):\n",
+    "        day = firstDay + timedelta(days = i)\n",
+    "        # Set year, month and and day here\n",
+    "        year, month, day = (day.year, day.month, day.day)\n",
+    "        currentPath = f's3://{bucketName}/{read_path}/year={year}/month='f'{month:02d}''/day='f'{day:02d}'\n",
```
Comment (gmfrasca) [inline Ray-Only.ipynb]: you only need the first 'f' prefix in an f-string, these additional ones make the string more confusing.

ex:  `f's3://{bucketName}/{read_path}/year={year}/month={month:02d}/day={day:02d}'`

File: Ray-Only.ipynb [hunk 5]
```diff
@@ -46,70 +86,459 @@
...
+    "        df = ray.data.read_parquet(paths=currentPath, filesystem=fs_pyarrow)\n",
+    "        # <Union dataframe> if not day 1.\n",
+    "        if day != 1:\n",
+    "            master_df = master_df.union(df) # Followed by unioning the dataframe each time\n",
+    "        else:\n",
+    "            df = master_df\n",
+    "\n",
+    "    # Code here:\n",
+    "    # <Writeback single parquet for the month>\n",
+    "    df.repartition(1).write_parquet(path=f's3://{bucketName}/{write_path}/year={year}/month='f'{month:02d}', filesystem=fs_pyarrow)\n",
```
Comment (gmfrasca) [inline Ray-Only.ipynb]: see above f-string comment

File: Ray-Only.ipynb [hunk 6]
```diff
@@ -46,70 +86,459 @@
...
+    "duration = lastDay - firstDay\n",
+    "#print(duration)\n",
+    "for _ in range(12):\n",
+    "    for i in range(duration.days):\n",
+    "        day = firstDay + timedelta(days = i)\n",
+    "        # Set year, month and and day here\n",
+    "        year, month, day = (day.year, day.month, day.day)        \n",
+    "        # Code here:\n",
+    "        # <Read dataframe>\n",
+    "        currentReadPath = f's3://{bucketName}/{read_path}/year={year}/month='f'{month:02d}''/day='f'{day:02d}'\n",
```
Comment (gmfrasca) [inline Ray-Only.ipynb]: see above f-string comment

File: Ray-Only.ipynb [hunk 7]
```diff
@@ -46,70 +86,459 @@
...
+    "        print(\"Reading from:\",currentReadPath)\n",
+    "        df = ray.data.read_parquet(paths=currentReadPath, filesystem=fs_pyarrow)\n",
+    "        # <Writeback single parquet for the day>\n",
+    "        currentWritePath = f's3://{bucketName}/{write_path}/year={year}/month='f'{month:02d}''/day='f'{day:02d}'\n",
```
Comment (gmfrasca) [inline Ray-Only.ipynb]: see above f-string comment

File: Ray-Only.ipynb [hunk 8]
```diff
@@ -46,70 +86,459 @@
...
+    "    #print(lastDay, firstDay)\n",
+    "    duration = lastDay - firstDay\n",
+    "    #print(duration)\n",
+    "    InitializeS3()\n",
+    "    for _ in range(12):\n",
+    "        for i in range(duration.days):\n",
+    "            day = firstDay + timedelta(days = i)\n",
+    "            # Set year, month and and day here\n",
+    "            year, month, day = (day.year, day.month, day.day)\n",
+    "            currentReadPath = f's3://{bucketName}/{read_path}/year={year}/month='f'{month:02d}''/day='f'{day:02d}'\n",
```
Comment (gmfrasca) [inline Ray-Only.ipynb]: see above f-string comment

File: Ray-Only.ipynb [hunk 9]
```diff
@@ -46,70 +86,459 @@
...
+    "            print(\"Reading from:\",currentReadPath)\n",
+    "            # Code here:\n",
+    "            # <Read dataframe>\n",
+    "            df = ray.data.read_parquet(paths=currentReadPath, filesystem=fs_pyarrow)\n",
+    "            # <Writeback single parquet for the day>\n",
+    "            currentWritePath = f's3://{bucketName}/{write_path}/year={year}/month='f'{month:02d}''/day='f'{day:02d}'\n",
```
Comment (gmfrasca) [inline Ray-Only.ipynb]: see above f-string comment

File: Ray-Only.ipynb (hunk 2 above)
Comment (Joeavaikath) [reply Ray-Only.ipynb:423]: Its actually a to-do, a block of code should go there. Remote function calls in Ray for code involving S3 fails, something I'm trying to do on the side.

File: Ray-Only.ipynb (hunk 3 above)
Comment (Joeavaikath) [reply Ray-Only.ipynb]: Done!

**Pull request merged**